drivers/flash/flash_stm32g4x.c

```diff
@@ -326,6 +320,15 @@ void flash_stm32_set_rdp_level(const struct device *dev, uint8_t level)
 {
 	flash_stm32_option_bytes_write(dev, FLASH_OPTR_RDP_Msk,
 				       (uint32_t)level << FLASH_OPTR_RDP_Pos);
+
+	/*
+	 * POR needed to reload option bytes, done at runtime by entering standby mode
+	 * /!\ make sure a wakeup source is configured!
+	 */
+	LL_PWR_SetPowerMode(LL_PWR_MODE_STANDBY);
+	LL_LPM_EnableDeepSleep();
+	/* enter SLEEP mode : WFE or WFI */
+	k_cpu_idle();
```

Comment:
```suggestion
	__disable_irq();
	__SEV();
	__WFE();
	__WFE();
```
otherwise entry in low-power state is not guaranteed (and even this sequence may not be 100% safe)

Reply:
This should go in common flash code if anywhere; PM needs something else (we **don't** want to *force* entry in low-power mode while doing PM, unlike here)